zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -88,40 +80,45 @@ private ZeebeRocksDbFactory(
   }
 
   private DBOptions createDefaultDbOptions(final List<AutoCloseable> closeables) {
-    final var statistics = new Statistics();
-    closeables.add(statistics);
-    statistics.setStatsLevel(StatsLevel.ALL);
+    final var dbOptions =
+        new DBOptions()
+            .setErrorIfExists(false)
+            .setCreateIfMissing(true)
+            .setParanoidChecks(true)
+            // 1 flush, 1 compaction
+            .setMaxBackgroundJobs(2)
+            // we only use the default CF
+            .setCreateMissingColumnFamilies(false)
+            // may not be necessary when WAL is disabled, but nevertheless recommended to avoid
+            // many small SST files
+            .setAvoidFlushDuringRecovery(true)
+            // fsync is called asynchronously once we have at least 4Mb
+            .setBytesPerSync(4 * 1024 * 1024L)
+            // limit the size of the manifest (logs all operations), otherwise it will grow
+            // unbounded
+            .setMaxManifestFileSize(256 * 1024 * 1024L)
+            // speeds up opening the DB
+            .setSkipStatsUpdateOnDbOpen(true)
+            // keep 1 hour of logs - completely arbitrary. we should keep what we think would be
+            // a good balance between useful for performance and small for replication
+            .setLogFileTimeToRoll(Duration.ofMinutes(30).toSeconds())
+            .setKeepLogFileNum(2)
+            // can be disabled when not profiling
+            .setStatsDumpPeriodSec(20);
```

Comment:
this should also be set only if statistics is enabled, or?

Reply:
ah makes probably sense :+1: